lib/videoroom/metrics_persistor.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Membrane.VideoRoom.MetricsPersistor do
+  use GenServer
+
+  alias Membrane.RTC.Engine.TimescaleDB.Reporter
+
+  # 1 second
+  @scrape_interval 1000
+
+  # 1 hour
+  @cleanup_interval 1000 * 60 * 60
+
+  @spec start(Reporter.reporter(), GenServer.options()) :: GenServer.on_start()
+  def start(reporter, options \\ []) do
+    do_start(:start, reporter, options)
+  end
+
+  @spec start_link(Reporter.reporter(), GenServer.options()) :: GenServer.on_start()
+  def start_link(reporter, options \\ []) do
+    do_start(:start_link, reporter, options)
+  end
+
+  defp do_start(function, reporter, options) do
+    apply(GenServer, function, [__MODULE__, reporter, options])
+  end
+
+  @spec child_spec(Keyword.t()) :: Supervisor.child_spec()
+  def child_spec(opts) do
+    {reporter, process_opts} = Keyword.pop(opts, :reporter, nil)
+
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [reporter, process_opts]}
+    }
+  end
```

Comment:
What's the point of overriding the child_spec defined by `use GenServer`? You can do the `Keyword.pop` inside the `do_start` function